From 186c79afa0a34e74c56ee6a8f9aab848532e6fb3 Mon Sep 17 00:00:00 2001
From: lfdevs <109842948+lfdevs@users.noreply.github.com>
Date: Fri, 30 Jan 2026 00:05:11 +0800
Subject: [PATCH 1/2] Deps: Linux arm64: Fix usd build error

Restrict Valgrind x86 assembly code to compile only on the x86_64 platform
---
 build_files/build_environment/cmake/usd.cmake | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/build_files/build_environment/cmake/usd.cmake b/build_files/build_environment/cmake/usd.cmake
index 0d28f4f7fa5..e8df43e808a 100644
--- a/build_files/build_environment/cmake/usd.cmake
+++ b/build_files/build_environment/cmake/usd.cmake
@@ -162,6 +162,9 @@ ExternalProject_Add(external_usd
     ${PATCH_CMD} -p 1 -d
       ${BUILD_DIR}/usd/src/external_usd <
       ${PATCH_DIR}/usd_3666_vulkan_amd.diff
+    # Fix ARM64 build error: Restrict Valgrind x86 assembly code to compile only on the x86_64 platform
+    COMMAND sed -i "s/defined(ARCH_OS_LINUX)/defined(ARCH_OS_LINUX) \\&\\& defined(__x86_64__)/"
+      ${BUILD_DIR}/usd/src/external_usd/pxr/exec/vdf/executorDataVector.cpp
   CMAKE_ARGS
     -DCMAKE_INSTALL_PREFIX=${LIBDIR}/usd
     -Wno-dev
-- 
2.47.3

